src/python_testing/TC_CLCTRL_2_1.py

```diff
@@ -0,0 +1,166 @@
+
+import logging
+
+import chip.clusters as Clusters
+from chip.testing.matter_testing import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+class TC_CLCTRL_2_1(MatterBaseTest):
+    async def read_closurecontrol_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.ClosureControl
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    def desc_TC_CLCTRL_2_1(self) -> str:
+        return "[TC-CLCTRL-2.1] Attributes with Server as DUT"
+
+    def steps_TC_CLCTRL_2_1(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep(2, "Read AttributeList attribute to determine supported attributes"),
+            TestStep(3, "Read CountdownTime attribute if supported"),
+            TestStep(4, "Read MainState attribute"),
+            TestStep(5, "Read CurrentErrorList attribute"),
+            TestStep(6, "Read OverallState attribute"),
+            TestStep(7, "Read OverallTarget attribute"),
+        ]
+        return steps
+
+    def pics_TC_CLCTRL_2_1(self) -> list[str]:
+        pics = [
+            "CLCTRL.S",
+        ]
+        return pics
+
+    @async_test_body
+    async def test_TC_CLCTRL_2_1(self):
+        
+        endpoint = self.get_endpoint(default=1)
+
+        # STEP 1: Commission DUT to TH (can be skipped if done in a preceding test)
+        self.step(1)
+        attributes = Clusters.ClosureControl.Attributes
+
+        # STEP 2: Read AttributeList attribute to determine supported attributes
+        self.step(2)
+        attribute_list = await self.read_closurecontrol_attribute_expect_success(endpoint=endpoint, attribute=attributes.AttributeList)
+        logging.info(f"Attribute list: {attribute_list}")
+        
+        # STEP 3: Read CountdownTime attribute if supported
+        self.step(3)
+        if attributes.CountdownTime.attribute_id in attribute_list:
+            countdown_time = await self.read_closurecontrol_attribute_expect_success(endpoint=endpoint, attribute=attributes.CountdownTime)
+            if countdown_time is not None:
+                asserts.assert_less_equal(countdown_time, 259200, "CountdownTime attribute is out of range")
+                asserts.assert_greater_equal(countdown_time, 0, "CountdownTime attribute is out of range")
+            logging.info(f"CountdownTime: {countdown_time}")
+        else:
+            logging.info("CountdownTime attribute not supported, skipping step")
+
+        # STEP 4: Read MainState attribute
+        self.step(4)
+        main_state = await self.read_closurecontrol_attribute_expect_success(endpoint=endpoint, attribute=attributes.MainState)
+        asserts.assert_less_equal(main_state, 7, "MainState attribute is out of range")
+        asserts.assert_greater_equal(main_state, 0, "MainState attribute is out of range")
```

Comment:
Python Scripts have access to the generated enums. We should use them as much as possible in the test validations. This is valid for the whole test.

`src/controller/python/chip/clusters/Objects.py`